src/Php83/Php83.php

```diff
@@ -82,4 +83,116 @@ public static function mb_str_pad(string $string, int $length, string $pad_strin
                 return mb_substr(str_repeat($pad_string, $leftPaddingLength), 0, $leftPaddingLength, $encoding).$string.mb_substr(str_repeat($pad_string, $rightPaddingLength), 0, $rightPaddingLength, $encoding);
         }
     }
+
+    public static function str_increment(string $string): string
+    {
+        if ('' === $string) {
+            throw new \ValueError('str_increment(): Argument #1 ($string) cannot be empty');
+        }
+
+        if (!\preg_match("/^[a-zA-Z0-9]+$/", $string)) {
+            throw new \ValueError('str_increment(): Argument #1 ($string) must be composed only of alphanumeric ASCII characters');
+        }
+
+        if (\is_numeric($string)) {
+            $offset = stripos($string, 'e');
+            if ($offset !== false) {
+                $char = $string[$offset];
+                $char++;
+                $string[$offset] = $char;
+                $string++;
+
+                switch ($string[$offset]) {
+                    case 'f':
+                        $string[$offset] = 'e';
+                        break;
+                    case 'F':
+                        $string[$offset] = 'E';
+                        break;
+                    case 'g':
+                        $string[$offset] = 'f';
+                        break;
+                    case 'G':
+                        $string[$offset] = 'F';
+                        break;
+                }
+
+                return $string;
+            }
+        }
+
+        return ++$string;
+    }
+
+    public static function str_decrement(string $string): string
+    {
+        if ('' === $string) {
+            throw new \ValueError('str_decrement(): Argument #1 ($string) cannot be empty');
+        }
+
+        if (!\preg_match("/^[a-zA-Z0-9]+$/", $string)) {
+            throw new \ValueError('str_decrement(): Argument #1 ($string) must be composed only of alphanumeric ASCII characters');
+        }
+
+        if (\preg_match('/\A(?:0[aA0]?|[aA])\z/', $string)) {
+            throw new \ValueError(sprintf('str_decrement(): Argument #1 ($string) "%s" is out of decrement range', $string));
+        }
+
+        if (!\in_array(substr($string, -1), ['A', 'a', '0'], true)) {
+            return join('', array_slice(str_split($string), 0, -1)) . chr(ord(substr($string, -1)) - 1);
+        }
+
+        $toConvert = strrev($string);
```

Comment:
it'd be great for performance and memory to get rid of the double strrev, doable?

Reply:
Only the first one I think